packages/components/src/font-size-picker/select.tsx

```diff
@@ -0,0 +1,74 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import CustomSelectControl from '../custom-select-control';
+import type {
+	FontSizePickerSelectProps,
+	FontSizePickerSelectOption,
+} from './types';
+import { splitValueAndUnitFromSize } from './utils';
+
+const FontSizePickerSelect = ( props: FontSizePickerSelectProps ) => {
+	const { fontSizes = [], value, size, onChange, onSelectCustom } = props;
+
+	const defaultOption: FontSizePickerSelectOption = {
+		key: 'default',
+		name: __( 'Default' ),
+		value: undefined,
+	};
+	const customOption: FontSizePickerSelectOption = {
+		key: 'custom',
+		name: __( 'Custom' ),
+	};
+	const options: FontSizePickerSelectOption[] = [
+		defaultOption,
+		...fontSizes.map( ( fontSize ) => {
+			const [ number ] = splitValueAndUnitFromSize( fontSize.size );
+			return {
+				key: fontSize.slug,
+				name: fontSize.name ?? fontSize.slug,
+				value: fontSize.size,
+				__experimentalHint: number,
```

Comment:
Minor nit: let's not call this variable `number`, since its type is actually a `string` (my brain refused to process this for a while 😅 )

```suggestion
			const [ parsedValue ] = splitValueAndUnitFromSize( fontSize.size );
			return {
				key: fontSize.slug,
				name: fontSize.name ?? fontSize.slug,
				value: fontSize.size,
				__experimentalHint: parsedValue,
```

Reply:
In https://github.com/WordPress/gutenberg/pull/44598 I changed this function to actually return a number. But yeah for now makes sense to name it what it is.